lib/dialects/abstract/query-generator.js

```diff
@@ -2670,7 +2670,7 @@ class QueryGenerator {
         }
 
         if (value.length > 0) {
-          return this._joinKeyValue(key, `(${value.map(item => this.escape(item, field)).join(', ')})`, comparator, options.prefix);
+          return this._joinKeyValue(key, `(${value.map(item => this.escape(item, field, { where: true })).join(', ')})`, comparator, options.prefix);
```

Comment:
If I remember right, there was some syntax required for IBM i to build this join that is triggered by the `where` option. I _thought_ about moving this to the IBM i query generator, but it felt like overkill for a single change to this large function. Trying to track down the test now.

Reply:
Tracked it down. There are queries where Db2 for i will not accept `NULL`, and instead required `cast(NULL as int)`. I have asked the developers of the database, and they were surprised by this as well, so I _think_ its a database bug. But anyways, I need a way to differentiate between `NULL` in these two types of queries:

```
SELECT "id", "createdAt", "updatedAt" FROM "Users" AS "User" WHERE "User"."id" IN (1, 2, cast(NULL as int))
```
and
```
SELECT "createdAt", "updatedAt", "deletedAt", "UserId", "ProjectId" FROM "UserProjects" AS "UserProjects" WHERE ("UserProjects"."deletedAt" IS NULL AND ("UserProjects"."UserId" = 2 AND "UserProjects"."ProjectId" IN (2, 1, 3))
```
I looked at the values passed to escape in those 2 calls to `escape`, and couldn't find anything that I could use to differentiate them. So I passed `where: true` as a flag to indicate that NULL should be cast in this case.

Reply:
Makes sense then. Is the issue with not accepting `NULL` documented somewhere public so we know when the casting can be removed?

Reply:
I don't believe it is really documented behavior just a bug. I also think it won't be fixed until a future release of IBM i, which will remain "in support" for 4-8 years. So I think it will have to remain for the foreseeable future, unless we can figure a different way to indicate this particular SQL structure.